### PR TITLE
Move data into separate json #2367

### LINKED ISSUE #2367: Lazy load data
Breaking away from a discussion about data size on #2297 so that I can have a chat with @shaneporter about how we load the data. 

### Today
The data is serialised into JSON and embedded in the html page. Some of the data sets are getting big (~8Mb in csv and 15Mb in html) and this makes the page slow to load. My experience is that the top of the page loads OK but then the metadata has to wait for all of the data to fill in.
### The future
A service will provide and API to get just the data you need.
### In between
As a stop gap solution can we think of ways to improve page loads when the data is big? I had two thoughts:

1. Move the JSON right to the very bottom of the html file so that the metadata loads.
2. As part of the build we could put the JSON in separate files, rather than embed it, and then you can send a request for it and render the chart when it arrives.
3. Same as 2 but we leave the headline data on the page so that we can quickly show a chart.

Any thoughts @shaneporter?

### PR DESCRIPTION
**Pull request by dougmet**
This PR changes the pages so that data is not embedded in the page, rather it is stored separately as a json file that is requested at load. This has the following benefits:

1. The page can carry on rendering while we wait for the data response.
2. We can compress the data greatly to improve transfer time.

The json is created by a python script, `scripts/build/build_json.py`.

Fixes #2367

### CODE REVIEW

**Comment by dougmet** [comment]
I've got the build to pass. The issue was that I developed in python 3.6 with pandas 0.22 and the build box has python 3.4 and pandas 0.13.1. A couple of syntax issues needed fixing. Also, the old `json.dump` didn't like numbers so I convert them on the fly.

File: scripts/build/build_json.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def nan_to_none(x):
 
 def dict_col_nan_to_none(d):
     """Take a dictionary of lists and replace all nans with None"""
-    out = {col: str([nan_to_none(x) for x in d[col]]) for col in d.keys()}
```
Comment (dougmet) [inline scripts/build/build_json.py]: I had to put this str in because python 3.4 couldn't cope turning numbers into JSON. What issue was it causing you?

**Comment by dougmet** [comment]
OK I fixed the build without needing further python libraries. I had done a very silly thing (I have a banging R-hangover)

**Comment by dougmet** [comment]
I realised I was writing out quoted numbers. But now the build is falling over on 10-b-1 for some reason.

**Comment by dougmet** [comment]
Oh god, writing the data out as JSON caused such a pain! Python 3.4 wouldn't serialise columns that are integers because they're numpy int64s, so I switched to the pandas json dumper, which works but for some reason it was double escaping forward slashes and Jekyll was getting upset about that. So I used a find and replace to remove those. 

Long story short it feels fragile and took ages to fix 😢

I'm pretty sure it's working now. The one cool thing about the script I wrote is that it now reads the data back in and checks it so it should be a good quality JSON dump.

**Comment by shaneporter** [comment]
The JSON seems good, and with 6356586, the UI is a little better too. Selections, chart and table are now hidden until the data is ready.

Overall page memory footprint is smaller, UI is snappier, and for those without JavaScript the pages will load considerably faster, particularly 4-2-1.

If you could take a look @dougmet that'd be good, but things seem ok to me.

**Comment by dougmet** [comment]
Awesome. I'll remove WIP review and merge.

**Comment by dougmet** [comment]
Looks good to me. Much better experience for pages with large data sets.

**Pull request merged**